### PR TITLE
Added Node Runlist api for infra proxy

### PR DESCRIPTION
**Pull request by kalroy**
Signed-off-by: Kallol Roy <karoy@progress.com>

### :nut_and_bolt: Description: What code changed, and why?

The Nodedetails api currently only provides the runlist of the node at the top level.
These runlists can have recipes and roles. The roles in the run list need to be expanded and the recipes also need to be given a position. 
Also the recipe/role need not be traversed multiple times since they are also considered once in the client runs.

We had a similar behavior available for Roles for Organization which can be reused for Nodes too.

The API route is:
`GET      ​/api​/v0​/infra​/servers​/{server_id}​/orgs​/{org_id}​/node​/{name}​/runlist​/{environment}`

The response format:
```
{
  "id": "string",
  "run_list": [
    {
      "type": "string",
      "name": "string",
      "version": "string",
      "skipped": true,
      "position": 0,
      "error": "string",
      "children": [
        null
      ]
    }
  ]
}
```
### :chains: Related Resources
#5056
### :athletic_shoe: How to Build and Test the Change

- Build the components:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
- Add data using 3rd internal steps https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
- Get Node details:
```
curl --location --request GET 'https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/nodes/chefnode2' 
--header 'api-token: ${api-token}'
```
- Get Node runlist
```
curl --location --request GET 'https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/node/chefnode2/runlist/_default'  --header 'api-token: ${api-token}'
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: api/external/infra_proxy/response/common.proto [hunk 1]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package chef.automate.api.infra_proxy.response;
+option go_package = "github.com/chef/automate/api/external/infra_proxy/response";
+message RunList {
```
Comment (kalroy) [inline api/external/infra_proxy/response/common.proto:5]: Created a common `RunList` structure to be used by Roles and Nodes.

File: api/external/infra_proxy/response/roles.proto [hunk 2]
```diff
@@ -45,23 +46,6 @@ message ExpandedRunList {
     repeated RunList run_list = 2;
 }
 
-message RunList {
```
Comment (kalroy) [inline api/external/infra_proxy/response/roles.proto:48]: Moved it to the `common.proto` file to be used by both `Node` and `Role` runlists

File: components/infra-proxy-service/server/roles.go [hunk 3]
```diff
@@ -361,79 +359,6 @@ func findRoleFromRoleList(name string, result *RoleListResult) *chef.Role {
 	return nil
 }
 
-func toResponseExpandedRunList(client *ChefClient, service *service.Service, runlist []string, cookbooks chef.EnvironmentCookbookResult, runlistCache RunListCache) ([]*response.RunList, error) {
```
Comment (kalroy) [inline components/infra-proxy-service/server/roles.go:364]: Moved it to the util module

File: api/external/infra_proxy/response/roles.proto (hunk 2 above)
Comment (rickmarry) [reply api/external/infra_proxy/response/roles.proto:48]: cool

**Review by rickmarry — APPROVED** [review]
thanks @kalroy!  I like the refactoring that you've done to make some of this reusable rather than having duplicate code

**Review by punitmundra — APPROVED** [review]
changes look fine.

**Pull request merged**